### PR TITLE
Add configurable overflow handling to FemtoFileHandler

### PR DESCRIPTION
**Pull request by leynos**
## Summary
Introduce configurable overflow policies for the FemtoFileHandler queue, expose them to Python and tests, and add documentation and usage examples.

## Testing
- `make fmt`
- `make lint`
- `make typecheck`
- `make test`

------
https://chatgpt.com/codex/tasks/task_e_686a978ac0348322846a35a49593f7c2

## Summary by Sourcery

Introduce configurable overflow policies for the FemtoFileHandler queue, expose them to Python and tests, and add documentation and usage examples.

New Features:
- Add OverflowPolicy enum to control how the handler behaves when its queue is full
- Expose new Python constructors for blocking and timeout overflow behaviors
- Provide a test-only with_writer_for_test constructor for injecting custom writers and policies
- Include a usage example of with_capacity_flush_timeout in the README

Enhancements:
- Centralize queue and flush behavior into a new with_capacity_flush_policy method
- Export OverflowPolicy in the library’s public API

Documentation:
- Document overflow policies and usage example in README and handler documentation

Tests:
- Add Rust tests for blocking and timeout policies using the test writer
- Add Python tests for blocking and timeout overflow behaviors

## Summary by Sourcery

Enable configurable overflow handling for FemtoFileHandler queue by introducing OverflowPolicy with Drop, Block, and Timeout options, centralizing handler setup into HandlerConfig and spawn_worker, exposing policy-aware constructors in both Rust and Python, adding test utilities and coverage, and updating documentation and Cargo configuration.

New Features:
- Introduce OverflowPolicy enum to control Drop, Block, or Timeout behavior when the queue is full
- Add HandlerConfig and TestConfig structs for configurable file handler setup and test injection
- Expose Python constructors for blocking and timeout-based capacity and flush policies
- Provide a test-only with_writer_for_test constructor for injecting custom writers and policies
- Export OverflowPolicy, HandlerConfig, and TestConfig in the public Rust API

Enhancements:
- Centralize handler construction and background worker logic into create_with_policy, with_capacity_flush_policy, and spawn_worker methods

Build:
- Add "test-util" feature and update Cargo.toml dev-dependencies for test support

Documentation:
- Update README with usage examples for timeout-based overflow handling
- Extend handler documentation to describe configurable overflow policies

Tests:
- Add Rust unit tests covering blocking and timeout overflow policies using TestConfig
- Add Python integration tests for blocking and timeout policies under various capacity and flush scenarios

### CODE REVIEW

File: rust_extension/src/file_handler.rs [hunk 1]
```diff
@@ -227,3 +339,69 @@ impl Drop for FemtoFileHandler {
...
+                        if writeln!(writer, "{msg}")
+                            .and_then(|_| writer.flush())
+                            .is_err()
+                        {
+                            warn!("FemtoFileHandler write error");
+                        } else {
+                            writes += 1;
+                            if flush_interval != 0
+                                && writes % flush_interval == 0
+                                && writer.flush().is_err()
```
[the remark this answers is not in the document]
Comment (leynos) [reply rust_extension/src/file_handler.rs:379]: @coderabbitai Please propose a fix for this issue and write prompt for an AI agent to implement it.

File: rust_extension/src/file_handler.rs [hunk 2]
```diff
@@ -102,7 +297,11 @@ impl FemtoFileHandler {
         P: AsRef<Path>,
         F: FemtoFormatter + Send + 'static,
     {
-        Self::with_capacity_flush_interval(path, formatter, capacity, 1)
+        let cfg = HandlerConfig {
+            capacity,
+            ..HandlerConfig::default()
+        };
+        Self::with_capacity_flush_policy(path, formatter, cfg)
```
[the remark this answers is not in the document]
Comment (leynos) [reply rust_extension/src/file_handler.rs:304]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

File: rust_extension/src/file_handler.rs [hunk 3]
```diff
@@ -102,7 +364,8 @@ impl FemtoFileHandler {
         P: AsRef<Path>,
         F: FemtoFormatter + Send + 'static,
     {
-        Self::with_capacity_flush_interval(path, formatter, capacity, 1)
+        let cfg = Self::build_config(capacity, None, OverflowPolicy::Drop);
+        Self::with_capacity_flush_policy(path, formatter, cfg)
```
[the remark this answers is not in the document]
Comment (leynos) [reply rust_extension/src/file_handler.rs:368]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

**Comment by leynos** [comment]
@sourcery-ai review
